cores/esp8266/uart.c

```diff
@@ -571,36 +645,17 @@ uart_uninit(uart_t* uart)
     if(uart == NULL)
         return;
 
-    uart_stop_isr(uart);
-
-    switch(uart->rx_pin) 
-    {
-    case 3:
-        pinMode(3, INPUT);
-        break;
-    case 13:
-        pinMode(13, INPUT);
-        break;
-    }
-
-    switch(uart->tx_pin) 
-    {
-    case 1:
-        pinMode(1, INPUT);
-        break;
-    case 2:
-        pinMode(2, INPUT);
-        break;
-    case 15:
-        pinMode(15, INPUT);
-        break;
+    if(uart->tx_enabled && (!gdbstub_has_uart_isr_control() || uart->uart_nr != UART0)) {
+        uart_disable_pin(uart->tx_pin);
     }
 
-    if(uart->rx_enabled)
-    {
-        uart_stop_isr(uart);
+    if(uart->rx_enabled) {
         free(uart->rx_buffer->buffer);
```

Comment:
Why are these two frees not conditional together with the uart_disable_pin and uart_stop_isr calls? Is the buffer not needed when gdb is active? Or does the isr keep firing after this uninit call for gbd?

Reply:
It indeed seems those free should go into is the gdb-has-uart test

Reply:
I've reworked the `uart_uninit` function to always stop the ISR (which doesn't really stop the ISR when GDB is active, it just tells GDB there is no IRQ callback to stuff received bytes into).  Will be in a push in a few mins.

The `free`s are correct as-is, GDB does NOT use any of the uart.c `struct uart_` state.